code/data_structures/linked_list/linked_list/linked_list.rb

```diff
@@ -0,0 +1,92 @@
+# Part of Cosmos by OpenGenus Foundation
+class Node
+	attr_accessor :data
+	attr_accessor :next
+
+	def initialize(data)
+		@data = data
+		@next = nil
+	end
+end
+
+class LinkedList
+	attr_accessor :head
+
+	def initialize
+		@head = nil
+	end
+
+	# Return the number of elements in the linked list
+	def size
+		current_node = @head
+		length = 0
+		until current_node.nil? do
+			length += 1
+			current_node = current_node.next
+		end
+		return length
+	end
+
+	# Insert a new node at end of linked list
+	def insert!(data)
```

Comment:
Consider renaming this function `add` and adding  a separate function called `insert` which takes a parameter of an index, and inserts an element at (or before) that position. The whole point of linked lists is to make insertion easy and fast at any location, so omitting that key function nullifies the whole point of the class.